File: NET_8/FoodCall.Domain/Exceptions/InvalidEntityException.cs
namespace FoodCall.Domain.Exceptions;

public class InvalidEntityException : DomainException
{
    public InvalidEntityException(string message) 
        : base(message)
    {
    }

    public InvalidEntityException(string entityName, string reason) 
        : base($"{entityName} inválido(a): {reason}")
    {
    }
}

File: Java/src/main/java/com/foodcall/domain/exceptions/InvalidEntityException.java
package com.foodcall.domain.exceptions;

public class InvalidEntityException extends DomainException {
    
    public InvalidEntityException(String message) {
        super(message);
    }
    
    public InvalidEntityException(String entityName, String reason) {
        super(String.format("%s inválido(a): %s", entityName, reason));
    }
}